lib/flags.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const { Uint64 } = require('./uint64');
+
+class Flags {
+  static from(...args) {
+    if (args.length > 64) {
+      throw new Error('Flags does not support more than 64 values');
+    }
+    const values = new Map(args.map((v, i) => [v, i]));
+    const masks = [];
+    for (let i = 0; i < args.length; i++) {
+      masks.push(new Uint64(1).shiftLeft(i));
+    }
+
+    class FlagsClass {
+      static from(...args) {
+        return new FlagsClass(...args);
+      }
+
+      constructor(...args) {
+        if (args[0] instanceof Uint64) {
+          this.value = new Uint64(args[0]);
+        } else {
+          this.value = new Uint64(0);
+          args.forEach(arg => this.set(arg));
+        }
+      }
+
+      has(key) {
+        return values.has(key);
+      }
+
+      get(key) {
+        return values.has(key) ?
+          Uint64.shiftRight(this.value, values.get(key)).toUint32() & 1 : -1;
+      }
+
+      set(key) {
+        if (values.has(key)) {
+          this.value.or(masks[values.get(key)]);
+        } else {
+          throw new TypeError('Flags instance does not have such key');
```

Comment:
You can invert the condition and leave only one `if` branch here and in a bunch of other methods:
```js
if (!values.has(key)) {
  throw new TypeError(/*...*/);
}
this.value.operation(/*...*/);
```

Reply:
Could you extract `get` and have it:
```js
const v = values.get(key);
if (v === undefined) {
  throw ...
}
this.value.and(masks[v]).not();
```